tangelo/linq/target/target_sympy.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2023 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from tangelo.linq import Circuit
+from tangelo.linq.target.backend import Backend
+from tangelo.linq.translator import translate_circuit as translate_c
+from tangelo.linq.translator import translate_operator
+
+
+class SympySimulator(Backend):
+
+    def __init__(self, n_shots=None, noise_model=None):
+        super().__init__(n_shots, noise_model)
+
+    def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, initial_statevector=None):
+        """Perform state preparation corresponding to the input circuit on the
+        target backend, return the frequencies of the different observables, and
+        either the statevector or None depending on the availability of the
+        statevector and if return_statevector is set to True. For this
+        simulator, symbolic expression are supported. Gates can have unspecified
+        parameters.
+
+        Args:
+            source_circuit (Circuit): A circuit in the abstract format to be
+                translated for the target backend.
+            return_statevector (bool): Option to return the statevector as well,
+                if available.
+            initial_statevector (array/matrix or sympy.physics.quantum.Qubit): A v
+                alid statevector in the format supported by the target backend.
+
+        Returns:
+            dict: A dictionary mapping multi-qubit states to their corresponding
+                frequency.
+            sympy.Matrix: The symbolic statevector, if requested
+                by the user (if not, set to None).
+        """
+
+        from sympy import simplify
```

Comment:
what made you choose to do the imports here ?

Reply:
When I tried to do like in the other Simulator like:
```python
...
def __init__(self, n_shots=None, noise_model=None):
    import sympy
    super().__init__(n_shots, noise_model)
    self.sympy = sympy
...
```
And working with the `self.sympy` attributes, I get weird behaviors with the sympy package:
```
Traceback (most recent call last):
  File "test_sympy.py", line 22, in <module>
    hist, vect = backend.simulate(circ, return_statevector=True)#, initial_statevector=np.array([[0.], [0.], [0.], [1.]]))
  File "/home/alex/Codes/Tangelo/tangelo/linq/target/backend.py", line 258, in simulate
    return self.simulate_circuit(source_circuit,
  File "/home/alex/Codes/Tangelo/tangelo/linq/target/target_sympy.py", line 61, in simulate_circuit
    python_statevector = self.sympy.physics.quantum.qubit.Qubit("0"*(source_circuit.width))
AttributeError: module 'sympy.physics.quantum' has no attribute 'qubit'
```

This is the way I got around to continue working on this.